src/notebooks/R Mia Examples/Comparative-Metagenomics.qmd

```diff
@@ -0,0 +1,852 @@
+```
+
+2. The table could be big, so let's look at a sample of it (`head`)
+
+```{r}
+#| output: false
+t(head(analyses_metadata_df))
+```
+
+## Convert to [TreeSummarizedExperiment object using mia](https://microbiome.github.io/)
+
+```{r}
+#| output: false
+tse <- getResult(mg, analyses_metadata_df$analysis_accession)
+```
+
+# Normalization, alpha diversity indices and taxonomic profiles visualization
+## Cleaning the OTUs matrix
+1. First, identify the total counts per sample and add them to the colData of our `tse` object.
+
+```{r}
+tse <- addPerCellQC(tse)
+```
+
+2. We then plot a histogram to identify outliers, so we can remove samples with low coverage.
+
+```{r}
+library_sie_histogram <- ggplot(colData(tse)) +
+  geom_histogram(aes(x = sum), color = "black", 
+                 fill = "gray", bins = 50) +
+  labs(x = "Library size", y = "Frequency (n)") + 
+  scale_x_log10(breaks = scales::trans_breaks("log10", function(x) 10^x), 
+                labels = scales::trans_format("log10", scales::math_format(10^.x))) +
+  theme_bw() +
+  # Remove the grid
+  theme(panel.grid.major = element_blank(), 
+        panel.grid.minor = element_blank(),
+        panel.border = element_blank(),
+        panel.background = element_blank(),
+        # Add y-axis line
+        axis.line = element_line(colour = "black"))
+```
+
+We can see that samples with number of reads below 10^1.6 seem to be outliers. Let’s filter out the outliers and plot a new histogram.
+
+```{r}
+filtered_data <- colData(tse)[colData(tse)$sum > 10^1.6, ]
+
+filtered_library_sie_histogram <- ggplot(filtered_data) +
+  geom_histogram(aes(x = sum), color = "black", 
+                 fill = "gray", bins = 50) +
+  labs(x = "Library size", y = "Frequency (n)") + 
+  scale_x_log10(breaks = scales::trans_breaks("log10", function(x) 10^x), 
+                labels = scales::trans_format("log10", scales::math_format(10^.x))) +
+  theme_bw() +
+  # Remove the grid
+  theme(panel.grid.major = element_blank(), 
+        panel.grid.minor = element_blank(),
+        panel.border = element_blank(),
+        panel.background = element_blank(),
+        # Add y-axis line
+        axis.line = element_line(colour = "black"))
+```
+
+2. Remove singletons. Singletons are OTUs of size one, meaning that only one read was assigned to that OTU. These very low-abundance OTUs could be biologically real (belonging to the rare biosphere ([1](#ref-lynch2015))), or they could be false positives due to sequencing artefacts. Singletons observed in only one sample are more likely to be artefacts, and it is good practice to remove them from the OTUs counts table to avoid artificially over-estimating the OTUs richness. You can find more discussion about this in [Robert Edgar’s blog](https://drive5.com/usearch/manual/singletons.html).
+
+```{r}
+# Calculate the total counts for each taxa across all samples in the filtered data
+taxa_sums_filtered <- rowSums(assay(tse)[, rownames(filtered_data)])
```

Comment:
Hmm filtering didnt affect rows, rowSums(assay(tse)) should be the same than rowSums(assay(tse)[, rownames(filtered_data)]) or rowSums(assay(tse.filtered))? 

I suggest that you operate with the tse object directly, so instead of rowSums(assay(tse)[, rownames(filtered_data)]) you would do something like rowSums(assay(tse.filtered))